src/RoadRegistry.BackOffice.Api/ZipArchiveWriters/ForEditor/RoadSegmentsToZipArchiveWriter.cs

```diff
@@ -38,16 +47,41 @@ public async Task WriteAsync(ZipArchive archive, EditorContext context, Cancella
                 RoadSegmentDbaseRecord.Schema
             );
             await using (var dbfEntryStream = dbfEntry.Open())
-            using (var dbfWriter =
-                new DbaseBinaryWriter(
-                    dbfHeader,
-                    new BinaryWriter(dbfEntryStream, _encoding, true)))
+            using (var dbfWriter = new DbaseBinaryWriter(dbfHeader, new BinaryWriter(dbfEntryStream, _encoding, true)))
             {
-                var dbfRecord = new RoadSegmentDbaseRecord();
-                foreach (var data in context.RoadSegments.OrderBy(_ => _.Id).Select(_ => _.DbaseRecord))
+                foreach (var batch in context.RoadSegments
+                    .OrderBy(_ => _.Id)
+                    .Select(_ => _.DbaseRecord)
+                    .AsEnumerable()
+                    .Batch(_zipArchiveWriterOptions.RoadSegmentBatchSize))
                 {
-                    dbfRecord.FromBytes(data, _manager, _encoding);
-                    dbfWriter.Write(dbfRecord);
+                    var dbfRecords = batch
+                        .Select(x =>
+                        {
+                            var dbfRecord = new RoadSegmentDbaseRecord();
+                            dbfRecord.FromBytes(x, _manager, _encoding);
+                            return dbfRecord;
+                        })
+                        .ToList();
+
+                    var cachedStreetNameIds = dbfRecords
+                        .Select(record => record.LSTRNMID.Value)
```

Comment:
Take into account that this may be `null` or `-8` / `-9` (depending on how we're filling those) - there's no need to fetch those from the cache. Some for `RSTRNMID`

Reply:
Never mind ... I now see the `.HasValue`